File: papers/ecoop2012/twitter.scala
val users = array("gkossakowski", "odersky",
  "adriaanm")
for (user <- users.parSuspendable) {
  console.log("fetching " + user)
  val tweets = fetchTweets(user)
  console.log("finished fetching " + user)
  for (t <- tweets)
    console.log("fetched " + t.text)
}
console.log("done")






|$\quad$|

File: papers/ecoop2012/twitter.js
var processed = 0
var users = ["gkossakowski", "odersky",
  "adriaanm"]
users.forEach(function (user) {
  console.log("fetching " + user)
  fetchTweets(user, function(data) {
    console.log("finished fetching " + user)
    data.forEach(function (tweet) {
      console.log("fetched " + tweet.text)
    })
    processed += 1
    if (processed == users.length) {
      console.log("done")
    }
  })
})
